src/main/java/weavers/siltarae/login/service/LoginService.java

```diff
@@ -0,0 +1,70 @@
+package weavers.siltarae.login.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import weavers.siltarae.global.exception.AuthException;
+import weavers.siltarae.global.exception.BadRequestException;
+import weavers.siltarae.global.exception.ExceptionCode;
+import weavers.siltarae.login.domain.GoogleProvider;
+import weavers.siltarae.login.domain.TokenProvider;
+import weavers.siltarae.login.dto.response.AccessTokenResponse;
+import weavers.siltarae.login.dto.response.TokenPair;
+import weavers.siltarae.login.dto.response.MemberInfoResponse;
+import weavers.siltarae.member.domain.Member;
+import weavers.siltarae.member.domain.repository.MemberRepository;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class LoginService {
+
+    private final MemberRepository memberRepository;
+    private final TokenProvider tokenProvider;
+    private final GoogleProvider googleProvider;
+
+    private final String BEARER_TYPE = "Bearer";
+
+    public TokenPair login(String socialType, String code) {
+        String authAccessToken = googleProvider.requestAccessToken(code);
+        MemberInfoResponse memberInfo = googleProvider.getMemberInfo(authAccessToken);
+
+        Member member = memberRepository.findByIdentifier(memberInfo.getIdentifier())
+                .orElseGet(() -> createMember(memberInfo));
+
+        return tokenProvider.createTokenPair(member.getId());
+    }
+
+    public AccessTokenResponse renewAccessToken(String authorizationHeader, String refreshToken) {
+        final String accessToken = getAccessTokenFromHeader(authorizationHeader);
+
+        if(!tokenProvider.isExpiredAccessAndValidRefresh(accessToken, refreshToken)) {
+            throw new AuthException(ExceptionCode.FAIL_TO_VALIDATE_TOKEN);
+        }
+
+        Long memberId = tokenProvider.getMemberIdFromAccessToken(accessToken);
+        String newAccessToken = tokenProvider.renewAccessToken(memberId);
+
+        return AccessTokenResponse.builder()
+                .accessToken(newAccessToken)
+                .build();
+    }
+
+    private String getAccessTokenFromHeader(String authorizationHeader) {
+        if(authorizationHeader==null || !authorizationHeader.startsWith(BEARER_TYPE)) {
+            throw new BadRequestException(ExceptionCode.INVALID_REQUEST);
+        }
+        return authorizationHeader.substring(BEARER_TYPE.length());
+    }
```

Comment:
해당 부분 tokenProvider 이 더 맞는 것 같아요 토큰을 활용하여 Bearer 을 제외한 엑세스토큰을 얻는 부분인 것 같아서요

Reply:
토큰을 활용하지는 않고 헤더에 "Bearer [accessToken]"으로 저장된 값을 Bearer 제외하고 파싱하는 부분입니다! 다시 보니 공백 제거하는 부분을 빼먹어서 아래와 같이 수정하려 합니다.

```suggestion
    private String getAccessTokenFromHeader(String authorizationHeader) {
        if(authorizationHeader==null || !authorizationHeader.startsWith(BEARER_TYPE)) {
            throw new BadRequestException(ExceptionCode.INVALID_REQUEST);
        }
        return authorizationHeader.substring(BEARER_TYPE.length()).trim();
    }
```

액세스 토큰을 추출하는 부분은 앞으로 '회원 전용 기능'을 사용할 때마다 필요할 것 같은데, 추후 Filter나 Interceptor를 사용한다면 해당 클래스로 이관될 것 같기도 해요🤔